chapter02/worksheets/solution2.1.7.9.scala

```diff
@@ -0,0 +1,29 @@
+/** author: Andreas Röhler */
+
+/** 
+  Exercise 2.1.7.9
+  Given a Seq[(String, Int)] showing a list of purchased items (where item names
+may repeat), compute a Map[String, Int] showing the total counts. So, for the input:
+Seq(("apple", 2), ("pear", 3), ("apple", 5), ("lemon", 2), ("apple", 3))
+
+the output must be Map("apple" -> 10, "pear" -> 3, "lemon" -> 2).
+Hint: use groupBy, map, sum.
+ */
+
+val a: Seq[(String, Int)] =  Seq(("pear", 2), ("apple", 2), ("apple", 6), ("pear", 1), ("lemon", 4), ("apple", 2), ("apple", 2))
+
+val b =  a.groupBy(s => s._1)
+
+val c = b.map{ case (x,y) => y}
+
+val e = c.map{ k=>k.map{ case x=>x._1} }.map{ x=>x(0)}
+
+val f = e.map{ case x => x.replaceAll("(\\w+)", "\"$1\"")}
+
+val g = c.map{ k=>k.map{ case x=>x._2}.sum }
+
+val result = f.zip(g).toMap
```

Comment:
Die Lösung ist richtig aber etwas kompliziert. 

Ich würde erstmals nicht die Anführungszeichen hinzufügen, dies habe ich nicht gemeint.

Scala gibt zwar so aus:

```scala
> List("abc", "def")
val res0: List[String] = List(abc, def)
```

ohne Anführungszeichen, aber die Strings `"abc"` und `"def"` bleiben Strings.

Ich habe also erwartet, wie in der Aufgabenstellung geschrieben, dass: `the output must be Map("apple" -> 10, "pear" -> 3, "lemon" -> 2)` und nicht `Map("\"apple\"" -> 10, ...)`

Zweitens, hier ist eine kürzere Sequenz von Berechnungen:

```scala
val a: Seq[(String, Int)] = ...
val b: Map[String, Seq[(String, Int)]] = a.groupBy(_._1)
val result: Map[String, Int] = b.map { case (name: String, pairs: Seq[(String, Int)]) => (name, pairs.map(_._2).sum) }
val expected = Map("apple" -> 12, "pear" -> 3, "lemon" -> 4)
assert( result == expected)
```